content/webapp/components/WorkDetails/WorkDetails.DownloadItem.tsx

```diff
@@ -0,0 +1,143 @@
+import { FunctionComponent } from 'react';
+import {
+  TransformedCanvas,
+  CustomContentResource,
+} from '@weco/content/types/manifest';
+import {
+  Body,
+  ContentResource,
+  ChoiceBody,
+  InternationalString,
+} from '@iiif/presentation-3';
+import {
+  getLabelString,
+  getFileSize,
+  isChoiceBody,
+} from '@weco/content/utils/iiif/v3';
+import styled from 'styled-components';
+import { font } from '@weco/common/utils/classnames';
+import Icon from '@weco/common/views/components/Icon/Icon';
+import { file, imageFile } from '@weco/common/icons';
+import { controlDimensions } from '@weco/content/components/ArchiveTree/ArchiveTree.helpers';
+
+export const DownloadTable = styled.table.attrs({ className: font('intr', 6) })`
+  position: relative;
+  height: ${`${controlDimensions.controlHeight}px`};
+  white-space: nowrap;
+  margin: 0;
+  width: 100%;
+
+  .icon {
+    position: relative;
+    top: 1px;
+    margin-right: 10px;
+  }
+
+  th,
+  td {
+    white-space: nowrap;
+    text-align: left;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    padding-right: 10px;
+  }
+
+  th:first-child {
+    padding-left: ${`${controlDimensions.controlWidth}px`};
+  }
+
+  th:nth-child(2),
+  td:nth-child(2) {
+    width: 120px;
+  }
+
+  th:nth-child(3),
+  td:nth-child(3) {
+    width: 60px;
+  }
+
+  th:last-child,
+  td:last-child {
+    width: 100px;
+    text-align: right;
+  }
+`;
+
+const getLabel = (item: Body) => {
+  if (typeof item !== 'string' && 'label' in item) {
+    return getLabelString(item.label as InternationalString);
+  } else {
+    return '';
+  }
+};
+
+const DownloadItem: FunctionComponent<{
+  canvas: TransformedCanvas | undefined;
+  item: (ContentResource | CustomContentResource | ChoiceBody) & {
+    format?: string;
+  };
+}> = ({ canvas, item }) => {
+  // If there is a choice then we only show the first one
+  const displayItem = (isChoiceBody(item) ? item.items[0] : item) as Body & {
+    format?: string;
+  };
+  const itemLabel = getLabel(displayItem);
+  const fileSize = canvas && getFileSize(canvas);
+  const format = displayItem.format;
+
+  if (typeof displayItem !== 'string') {
+    return (
+      <DownloadTable>
+        {/* TODO would like to put this back but it casues spacing issues in Safari */}
```

Comment:
Should this be addressed before merge?

Reply:
i don't think it's a deal breaker and I spent quite a lot of time trying to fix it. I'll create a separate issue for this